androidplot-core/src/main/java/com/androidplot/util/FastNumber.java

```diff
@@ -64,6 +64,32 @@ public double doubleValue() {
         return doublePrimitive;
     }
 
+    /**
+     * To be equal, two instances must both be instances of {@link FastNumber}. The inner {@link
+     * #number} field must also be a common type. Numbers which are mathematically equal are not
+     * necessarily equal. This keeps with the java implementation of common Number classes where for
+     * instance {@code new Integer(0).equals(new Double(0))} returns {@code false}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FastNumber that = (FastNumber) o;
+
+        return number != null ? number.equals(that.number) : that.number == null;
+
+    }
+
+    @Override
+    public int hashCode() {
+        return number != null ? number.hashCode() : 0;
```

Comment:
>This would result in the following being true:  
`new FastNumber(0).hashCode() == new FastNumber(null).hashCode()`

I figured this is okay as `hashCode` [is not required to produce distinct results for unequal objects](https://developer.android.com/reference/java/lang/Object.html#hashCode()). But never the less, we could add null checks to `FastNumber`.